### ISSUE TITLE
TRTorch on Jetson Nano

### ISSUE THREAD

**Issue by Michael-Equi** [NVIDIA/TRTorch#70]
Is there a build process outlined for getting this library built on a Jetson Nano or another ARM64 platform (TX2, Xavier, etc.)? The Bazel WORKSPACE seems rather dependent on X86 binaries and since I am new to Bazel I am having a hard time figuring out how to get it to build on Jetson Nano that already has TensorRT, CUDA, cuDNN, and libtorch/pytorch installed. Thanks

**Comment by narendasan** [comment]
Try using the local dependencies path for bringing sources for cuDNN and TensorRT. https://nvidia.github.io/TRTorch/tutorials/installation.html#building-using-locally-installed-cudnn-tensorrt.

If you have a libtorch build for Jetson, you can replace the `http_archive` with a similar `local_repository` target. If you are building for Python or are using the PyTorch distribution of `libtorch` (the one that comes with the wheel file) replace the `pre_cxx11_abi` libtorch target with something like this.

```
new_local_repository(
    name = "libtorch_pre_cxx11_abi",
    path = ".../site-packages/torch",
    build_file = "@//third_party/libtorch:BUILD",
)
```
otherwise replace the CXX11 ABI target with the path to your library like this:
```
new_local_repository(
    name = "libtorch",
    path = "<PATH TO LIBTORCH ROOT>",
    build_file = "@//third_party/libtorch:BUILD",
)
```

We are working on a better workflow, including hopefully cross compilation support but it is lower priority than other tasks right now.

**Comment by Michael-Equi** [comment]
Ah, finally got it working. I made those changes and then had to change some of the BUILD files for the third party libraries so it would find them correctly. I have all those edits made on a branch in my fork just in case anyone else would like to use them until cross compilation is supported in a cleaner way. https://github.com/BytesRobotics/TRTorch/tree/jetson-nano

**Comment by suyashhchougule** [comment]
> Ah, finally got it working. I made those changes and then had to change some of the BUILD files for the third party libraries so it would find them correctly. I have all those edits made on a branch in my fork just in case anyone else would like to use them until cross compilation is supported in a cleaner way. https://github.com/BytesRobotics/TRTorch/tree/jetson-nano

Hi, Did you had to switch to tensorrt 8.0 on jetson in order to install TRTorch ?

**Comment by narendasan** [comment]
Your TensorRT version should be determined by your Jetpack version. By adding the jetpack version as a flag you can use the previous version (4.5) if you are still using it. https://nvidia.github.io/TRTorch/tutorials/installation.html#building-natively-on-aarch64-jetson
